langchain/text_splitter.py

```diff
@@ -46,3 +37,72 @@ def split_text(self, text: str) -> List[str]:
             total += len(d)
         docs.append(self._separator.join(current_doc))
         return docs
+
+
+class CharacterTextSplitter(TextSplitter):
+    """Implementation of splitting text that looks at characters."""
+
+    def __init__(
+        self, separator: str = "\n\n", chunk_size: int = 4000, chunk_overlap: int = 200
+    ):
+        """Create a new CharacterTextSplitter."""
+        super(CharacterTextSplitter, self).__init__(
+            separator, chunk_size, chunk_overlap
+        )
+        self._separator = separator
+
+    def split_text(self, text: str) -> List[str]:
+        """Split incoming text and return chunks."""
+        # First we naively split the large input into a bunch of smaller ones.
+        splits = text.split(self._separator)
+        return self._merge_splits(splits)
+
+
+class NLTKTextSplitter(TextSplitter):
+    """Implementation of splitting text that looks at sentences using NLTK."""
+
+    def __init__(
+        self, separator: str = "\n\n", chunk_size: int = 4000, chunk_overlap: int = 200
+    ):
+        """Initialize the NLTK splitter."""
+        super(NLTKTextSplitter, self).__init__(separator, chunk_size, chunk_overlap)
+        try:
+            from nltk.tokenize import sent_tokenize
+
+            self._tokenizer = sent_tokenize
+        except ImportError:
+            raise ImportError(
+                "NLTK is not installed, please install it with " "`pip install nltk`. "
+            )
+
+    def split_text(self, text: str) -> List[str]:
+        """Split incoming text and return chunks."""
+        # First we naively split the large input into a bunch of smaller ones.
+        splits = self._tokenizer(text)
+        return self._merge_splits(splits)
+
+
+class SpacyTextSplitter(TextSplitter):
+    """Implementation of splitting text that looks at sentences using Spacy."""
+
+    def __init__(
+        self,
+        separator: str = "\n\n",
+        pipeline: str = "en_core_web_sm",
+        chunk_size: int = 4000,
+        chunk_overlap: int = 200,
+    ):
+        """Initialize the spacy text splitter."""
+        super(SpacyTextSplitter, self).__init__(separator, chunk_size, chunk_overlap)
+        try:
+            import spacy
+        except ImportError:
+            raise ImportError(
+                "Spacy is not installed, please install it with " "`pip install spacy`."
```

Comment:
nit: think the formatter did this, but this is two strings (...with " "`pip...) but can just be one

Reply:
done